db/db.go

```diff
@@ -26,32 +27,86 @@ type SQLClient interface {
 	InitSchema() error
 	InsertUsageEventList(data *cf.UsageEventList, tableName string) error
 	FetchLastGUID(tableName string) (string, error)
+	Exec(query string, args ...interface{}) (sql.Result, error)
+	Prepare(query string) (*sql.Stmt, error)
 	QueryJSON(q string, args ...interface{}) io.Reader
 	QueryRowJSON(q string, args ...interface{}) io.Reader
 	UpdateViews() error
-	Close() error
+	BeginTx() (SQLClient, error)
+	Rollback() error
+	Commit() error
+}
+
+type SQLConn interface {
+	QueryRow(query string, args ...interface{}) *sql.Row
+	Query(query string, args ...interface{}) (*sql.Rows, error)
+	Exec(query string, args ...interface{}) (sql.Result, error)
+	Prepare(query string) (*sql.Stmt, error)
 }
 
 // PostgresClient is the Postgres DB client for handling usage event queries
 type PostgresClient struct {
-	Conn *sql.DB
+	db   *sql.DB
+	tx   *sql.Tx
+	Conn SQLConn
 }
 
 // NewPostgresClient creates a new Postgres client
 func NewPostgresClient(connectionString string) (*PostgresClient, error) {
-	conn, err := sql.Open("postgres", connectionString)
+	db, err := sql.Open("postgres", connectionString)
 	if err != nil {
 		return nil, err
 	}
 	pc := &PostgresClient{
-		Conn: conn,
+		db:   db,
+		tx:   nil,
+		Conn: db,
 	}
 	return pc, nil
 }
 
+func (pc *PostgresClient) BeginTx() (SQLClient, error) {
```

Comment:
PostgresClient is becoming a God Objecty ... is it a Tx ... it's a DB

Think this is a code smell that the "methods" on this type should just be utility functions or something.